tests/unit/igraph_cohesion.c

```diff
@@ -0,0 +1,44 @@
+/*
+   IGraph library.
+   Copyright (C) 2021  The igraph development team <igraph@igraph.org>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include <igraph.h>
+#include <assert.h>
+int main() {
+
+    igraph_t g;
+    igraph_integer_t value;
+    igraph_bool_t checks = 1;
+
+    igraph_small(&g, 7, IGRAPH_DIRECTED,
+                 0, 1, 0, 2, 1, 2, 1, 3, 2, 4, 3, 4, 3, 5, 4, 5, 1, 6, 6, 3, 5, 0, -1);
+
+    igraph_cohesion(&g, &value, checks);
+
+    assert(value == 1);
+
```

Comment:
The graph needs to be destroyed here because creating a new one in its place. This is the cause of the test failure.